pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -55,7 +57,15 @@ def __init__(
             raise MisconfigurationException(f"`precision='bf16'` does not use a scaler, found {scaler}.")
         self.precision = precision
         self.device = device
-        self.scaler = scaler
+        self._scaler = scaler
+
+    @property
+    def scaler(self) -> Optional["GradScaler"]:
+        return self._scaler
+
+    @property
+    def amp_backend(self) -> Optional[LightningEnum]:
```

Comment:
Neat !

Reply:
We already have `backend` defined above the `__init__`

Reply:
But still, what's the motivation for `amp_backend` over `backend`? Shouldn't we have just one, whichever it is?

Reply:
backend is the outlier here, not amp_backend (the same name of the Trainer argument). We should revisit how these enums are used internally. This is touches several files outside the scope of this PR, so I suggest to do keep it like this for now. 
I will make a PR

Reply:
Part of the backend story is imo this simplification we need to do: https://github.com/PyTorchLightning/pytorch-lightning/issues/10821
There is my proposal how to differentiate the public attribute the user sees from what we use internally.

Reply:
Oh nice! There is already a issue we can discuss the trade-offs, thanks Adrian!